pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -286,3 +295,218 @@ func filterInformerActions(actions []core.Action) []core.Action {
 
 	return ret
 }
+
+// TestOverlappingDeployments ensures that an overlapping deployment will not be synced by
+// the controller.
+func TestOverlappingDeployments(t *testing.T) {
+	f := newFixture(t)
+	now := metav1.Now()
+	later := metav1.Time{Time: now.Add(time.Minute)}
+
+	foo := newDeployment("foo", 1, nil, nil, nil, map[string]string{"foo": "bar"})
+	foo.CreationTimestamp = now
+	bar := newDeployment("bar", 1, nil, nil, nil, map[string]string{"foo": "bar"})
+	bar.CreationTimestamp = later
```

Comment:
Should we use `SelectorUpdateAnnotation` instead of `CreationTimestamp`? (We can keep CreationTimestamp in this test, but make the older one's selector updated last)